flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/runtime/aggregate/AggregateMapFunction.scala

```diff
@@ -44,7 +44,7 @@ class AggregateMapFunction[IN, OUT](
 
   override def map(value: IN): OUT = {
     
-    val input = value.asInstanceOf[Row]
+    val input = value.asInstanceOf[Product]
```

Comment:
Currently aggregates do only support `Row`, because the aggregate code is not generated yet.
`DataSetAggregate` enforces `Row` as input type (see `DataSetAggregate` line 99), so `value.asInstanceOf[Row]` should be safe. 

Did you observe a problem with this cast?